### PR TITLE
Hooks hw2 part3

### PR DESCRIPTION
**Pull request by siarotchi**

### CODE REVIEW

File: src/components/todoClass/Form.js [hunk 1]
```diff
@@ -5,8 +5,12 @@ const Form = ({ handleEnter, addTask, inputChange }) => {
   const context = useContext(ContextClass);
   const { value, changeState, tasks } = context;
 
+  const handleAdd = (e) => {
+    e.preventDefault();
+    addTask(value, tasks, changeState);
+  };
   return (
-    <div className="task-input">
+    <form className="task-input">
```
Comment (eduardIliasenco-ui) [inline src/components/todoClass/Form.js:13]: Please use onSubmit

File: src/components/todoClass/Form.js [hunk 2]
```diff
@@ -15,14 +19,10 @@ const Form = ({ handleEnter, addTask, inputChange }) => {
         onKeyPress={(event) => handleEnter(event, tasks, changeState)}
```
Comment (eduardIliasenco-ui) [inline src/components/todoClass/Form.js:19]: you don't need to handle onEnter

File: src/components/todoClass/Form.js [hunk 3]
```diff
@@ -15,14 +19,10 @@ const Form = ({ handleEnter, addTask, inputChange }) => {
...
         value={value}
       ></input>
-      <button
-        onClick={() => addTask(value, tasks, changeState)}
-        type="submit"
-        className="btn btn-primary text-capitalize"
-      >
+      <button onClick={handleAdd} type="submit" className="btn btn-primary text-capitalize">
```
Comment (eduardIliasenco-ui) [inline src/components/todoClass/Form.js:22]: onClick={handleAdd} move that login into onSubmit

File: src/components/todoContext/TodoProvider.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ class TodoProvider extends React.Component {
   render() {
     const todoContextValue = {
       ...this.state,
-      changeState: (e) => this.setState(e),
+      changeState: (e) => this.setState(e), // this.setState
```
Comment (eduardIliasenco-ui) [inline src/components/todoContext/TodoProvider.js:13]: changeState: this.setState

File: src/components/todoFc/FormFc.js [hunk 5]
```diff
@@ -1,35 +1,73 @@
...
+import React, { useEffect, useState, useRef, forwardRef, useImperativeHandle } from "react";
 
-const FormFc = ({ inputChange, handleEnter, addTask }, ref) => {
-  const { value } = useContext(ContextFc);
+const FormFc = ({ addTask, taskIsBeindEdited, cleanEdit }, ref) => {
   const inputRef = useRef();
+  const [value, setValue] = useState("");
+
+  useEffect(() => {
+    if (taskIsBeindEdited) updateNewTitle(taskIsBeindEdited.title);
```
Comment (eduardIliasenco-ui) [inline src/components/todoFc/FormFc.js:8]: !! Used before it was defined

File: src/components/todoFc/FormFc.js [hunk 6]
```diff
@@ -1,35 +1,73 @@
...
+  const onCancelEdit = (e) => {
+    e.preventDefault();
+    addTask(taskIsBeindEdited.title);
+    setValue("");
+    cleanEdit();
+  };
+
   return (
-    <div className="task-input">
+    <form className="task-input">
```
Comment (eduardIliasenco-ui) [inline src/components/todoFc/FormFc.js:52]: Same here.... use onSubmit

File: src/components/todoFc/FormFcReducer.js [hunk 7]
```diff
@@ -26,6 +39,15 @@ const FormFcReducer = () => {
...
+    e.preventDefault();
+    addTask(setTaskIsBeingEdited.title);
+    setValue("");
+    cleanEdit();
+  };
+
+  console.log({ setTaskIsBeingEdited });
+
   return (
     <form className="task-input">
```
Comment (eduardIliasenco-ui) [inline src/components/todoFc/FormFcReducer.js:52]: Use onSubmit

File: src/pages/ToDo.js [hunk 8]
```diff
@@ -71,6 +71,7 @@ class Todo extends React.Component {
 
   handleEnter = (event, tasks) => {
```
Comment (eduardIliasenco-ui) [inline src/pages/ToDo.js:72]: You don't need to handle onEnter

File: src/pages/ToDoFc.js [hunk 9]
```diff
@@ -73,7 +68,15 @@ const TodoFc = () => {
       <div className="todo-container" style={{ height: `${containerHeight}px` }}>
         <h3 className="todo-header">ToDo Height: {containerHeight}</h3>
         <h1 className="todo-header">ToDo List: </h1>
-        <FormFc ref={inputRef} addTask={addTask} inputChange={inputChange} handleEnter={handleEnter} />
+        <FormFc
+          ref={inputRef}
```
Comment (eduardIliasenco-ui) [inline src/pages/ToDoFc.js:72]: How are you using inputRef here?

Comment (siarotchi) [reply src/pages/ToDoFc.js:72]: for setContainerHeight of container

**Pull request merged**